conjure-java-core/src/main/java/com/palantir/conjure/java/services/dialogue/AsyncGenerator.java

```diff
@@ -51,14 +51,20 @@
 import java.util.List;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.function.Function;
 import javax.lang.model.element.Modifier;
 
 public final class AsyncGenerator {
 
+    private final Function<com.palantir.conjure.spec.TypeName, TypeDefinition> typeNameResolver;
     private final ParameterTypeMapper parameterTypes;
     private final ReturnTypeMapper returnTypes;
 
-    public AsyncGenerator(ParameterTypeMapper parameterTypes, ReturnTypeMapper returnTypes) {
+    public AsyncGenerator(
+            Function<com.palantir.conjure.spec.TypeName, TypeDefinition> typeNameResolver,
```

Comment:
can I vote for a class/interface for this? lets us more easily identify if it's used elsewhere in the codebase and dedupe I think